src/Weather/Skysight/CDFDecoder.cpp

```diff
@@ -220,9 +220,9 @@ bool CDFDecoder::Decode() {
   size_t lat_size = data_file.get_dim("lat")->size();
   size_t lon_size = data_file.get_dim("lon")->size();
 
-  double lat_vals[lat_size];
-  double lon_vals[lon_size];
-  double var_vals[lat_size][lon_size];
+  double *lat_vals = new double[lat_size];
```

Comment:
*No* to manual memory management! This isn't exception-safe.
I don't even understand this commit. It fixes a bug from a previous commit in this PR - so this PR contains known-bad commits, but I don't want to read known-bad code.

Reply:
I am glad to integrate this patch into the previous commit. The reason why I separate is tp clearly indicate the previous commit is written by jfwells and I fix its bug.

I am not clear on manual memory management. You mean it should be run within try-catch?

Reply:
No, use RAII to free the memory. For example by using `std::unique_ptr<>` or our own `AllocatedArray` class.
Btw. the old code wasn't even legal C++; variable-size arrays are legal in C, but not in C++. This has only ever worked because GCC is sloppy about allowing (bad) C features in C++.

Reply:
convert to AllocatedArray.